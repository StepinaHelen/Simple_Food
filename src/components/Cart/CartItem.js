import { useContext } from "react";
import Icons from "../SvgComponent/SvgComponent";
import CartContext from "../../store/cart-context";
import { Item, WraperItem, AmountWrapper, BtnWrapper } from "./CartStyles";

const CartItem = (props) => {
  const cartContext = useContext(CartContext);
  const deleteFromCartHandler = () => {
    cartContext.removeItem(props.id);
  };

  return (
    <Item>
      <img src={props.img} alt={props.name} />
      <WraperItem>
        <h2>{props.title}</h2>
        <p>Price: {props.price}$</p>
        <AmountWrapper>
          <p>Amount: {props.amount}</p>
        </AmountWrapper>
        <BtnWrapper onClick={deleteFromCartHandler}>
          <Icons name="trash" classes={"trash"} />
        </BtnWrapper>
      </WraperItem>
    </Item>
  );
};

export default CartItem;
