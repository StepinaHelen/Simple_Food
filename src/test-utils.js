import { render } from "@testing-library/react";
import { QueryClientProvider, QueryClient } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { baseTheme } from "./styles/theme";
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={baseTheme}>
      <QueryClientProvider contextSharing={true} client={queryClient}>
        <BrowserRouter>{children}</BrowserRouter>
      </QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
