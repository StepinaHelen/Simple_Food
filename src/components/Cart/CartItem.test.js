import { render, screen, fireEvent } from "../../test-utils";
import userEvent from "@testing-library/user-event";
import CartItem from "./CartItem";

import CartContext from "../../store/cart-context";

const mockedCartLists = {
  id: 15,
  title: "Noodles with shrimp",
  price: 25,
  category: "noodles",
  amount: 1,
  img: "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg",
};

describe("test CartItem", () => {
  it("CartItem renders", () => {
    render(<CartItem {...mockedCartLists} />);
    expect(screen.getByText("Noodles with shrimp")).toBeInTheDocument();
    const img = screen.getByRole("img");
    expect(img.src).toContain(
      "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg"
    );
  });

  it("CartItem renders without data", () => {
    render(<CartItem />);
    expect(screen.queryByText("Noodles with shrimp")).toBeNull();
  });

  it("CartItem snapshot", () => {
    const view = render(<CartItem {...mockedCartLists} />);
    expect(view).toMatchSnapshot();
  });

  it("CartItem snapshot with empty data", () => {
    const view = render(<CartItem />);
    expect(view).toMatchSnapshot();
  });

  it("click on the delete button", async () => {
    const removeItem = jest.fn();
    render(
      <CartContext.Provider
        value={{
          removeItem,
        }}
      >
        <CartItem />
      </CartContext.Provider>
    );
    const deleteBtn = await screen.findByTestId("delete-btn");
    userEvent.click(deleteBtn);
    expect(removeItem).toHaveBeenCalledTimes(1);
  });
});
