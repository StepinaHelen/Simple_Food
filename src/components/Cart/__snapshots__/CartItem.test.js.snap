// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CartItem CartItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-bottom: 20px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c1 h2 {
  font-weight: 500;
  font-size: 40px;
  line-height: 47px;
  margin: 0 0 10px 0;
}

.c1 p {
  font-weight: 500;
  font-size: 28px;
  line-height: 33px;
  margin: 0 0 10px 0;
}

.c2 p {
  margin: 0;
}

.c3 {
  text-align: right;
}

.c3 .trash {
  width: 35px;
  height: 44px;
}

@media screen and (min-width:768px) {
  .c0 img {
    width: 330px;
    max-height: 199px;
  }
}

@media screen and (min-width:1200px) {
  .c0 img {
    max-height: 255px;
    width: 39%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding: 40px 32px;
    width: 61%;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    max-height: 199px;
    padding: 20px;
  }

  .c1 h2 {
    font-size: 28px;
    line-height: 33px;
    margin: 0 0 20px 0;
  }

  .c1 p {
    font-size: 20px;
    line-height: 23px;
    margin: 0 0 20px 0;
  }
}

@media screen and (min-width:1200px) {
  .c1 h2 {
    font-weight: 500;
    font-size: 40px;
    line-height: 47px;
  }

  .c1 p {
    font-weight: 500;
    font-size: 28px;
    line-height: 33px;
  }
}

<body>
    <div>
      <li
        class="c0"
      >
        <img
          src="https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg"
        />
        <div
          class="c1"
        >
          <h2>
            Noodles with shrimp
          </h2>
          <p>
            Price: 
            25
            $
          </p>
          <div
            class="c2"
          >
            <p>
              Amount: 
              1
            </p>
          </div>
          <div
            class="c3"
            data-testid="delete-btn"
          >
            <svg
              class="trash"
            >
              <use
                xlink:href="sprite.svg#trash"
              />
            </svg>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": .c0 {
  margin-bottom: 20px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c1 h2 {
  font-weight: 500;
  font-size: 40px;
  line-height: 47px;
  margin: 0 0 10px 0;
}

.c1 p {
  font-weight: 500;
  font-size: 28px;
  line-height: 33px;
  margin: 0 0 10px 0;
}

.c2 p {
  margin: 0;
}

.c3 {
  text-align: right;
}

.c3 .trash {
  width: 35px;
  height: 44px;
}

@media screen and (min-width:768px) {
  .c0 img {
    width: 330px;
    max-height: 199px;
  }
}

@media screen and (min-width:1200px) {
  .c0 img {
    max-height: 255px;
    width: 39%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding: 40px 32px;
    width: 61%;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    max-height: 199px;
    padding: 20px;
  }

  .c1 h2 {
    font-size: 28px;
    line-height: 33px;
    margin: 0 0 20px 0;
  }

  .c1 p {
    font-size: 20px;
    line-height: 23px;
    margin: 0 0 20px 0;
  }
}

@media screen and (min-width:1200px) {
  .c1 h2 {
    font-weight: 500;
    font-size: 40px;
    line-height: 47px;
  }

  .c1 p {
    font-weight: 500;
    font-size: 28px;
    line-height: 33px;
  }
}

<div>
    <li
      class="c0"
    >
      <img
        src="https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg"
      />
      <div
        class="c1"
      >
        <h2>
          Noodles with shrimp
        </h2>
        <p>
          Price: 
          25
          $
        </p>
        <div
          class="c2"
        >
          <p>
            Amount: 
            1
          </p>
        </div>
        <div
          class="c3"
          data-testid="delete-btn"
        >
          <svg
            class="trash"
          >
            <use
              xlink:href="sprite.svg#trash"
            />
          </svg>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test CartItem CartItem snapshot with empty data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-bottom: 20px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c1 h2 {
  font-weight: 500;
  font-size: 40px;
  line-height: 47px;
  margin: 0 0 10px 0;
}

.c1 p {
  font-weight: 500;
  font-size: 28px;
  line-height: 33px;
  margin: 0 0 10px 0;
}

.c2 p {
  margin: 0;
}

.c3 {
  text-align: right;
}

.c3 .trash {
  width: 35px;
  height: 44px;
}

@media screen and (min-width:768px) {
  .c0 img {
    width: 330px;
    max-height: 199px;
  }
}

@media screen and (min-width:1200px) {
  .c0 img {
    max-height: 255px;
    width: 39%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding: 40px 32px;
    width: 61%;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    max-height: 199px;
    padding: 20px;
  }

  .c1 h2 {
    font-size: 28px;
    line-height: 33px;
    margin: 0 0 20px 0;
  }

  .c1 p {
    font-size: 20px;
    line-height: 23px;
    margin: 0 0 20px 0;
  }
}

@media screen and (min-width:1200px) {
  .c1 h2 {
    font-weight: 500;
    font-size: 40px;
    line-height: 47px;
  }

  .c1 p {
    font-weight: 500;
    font-size: 28px;
    line-height: 33px;
  }
}

<body>
    <div>
      <li
        class="c0"
      >
        <img />
        <div
          class="c1"
        >
          <h2 />
          <p>
            Price: 
            $
          </p>
          <div
            class="c2"
          >
            <p>
              Amount: 
            </p>
          </div>
          <div
            class="c3"
            data-testid="delete-btn"
          >
            <svg
              class="trash"
            >
              <use
                xlink:href="sprite.svg#trash"
              />
            </svg>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": .c0 {
  margin-bottom: 20px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c1 h2 {
  font-weight: 500;
  font-size: 40px;
  line-height: 47px;
  margin: 0 0 10px 0;
}

.c1 p {
  font-weight: 500;
  font-size: 28px;
  line-height: 33px;
  margin: 0 0 10px 0;
}

.c2 p {
  margin: 0;
}

.c3 {
  text-align: right;
}

.c3 .trash {
  width: 35px;
  height: 44px;
}

@media screen and (min-width:768px) {
  .c0 img {
    width: 330px;
    max-height: 199px;
  }
}

@media screen and (min-width:1200px) {
  .c0 img {
    max-height: 255px;
    width: 39%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding: 40px 32px;
    width: 61%;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    max-height: 199px;
    padding: 20px;
  }

  .c1 h2 {
    font-size: 28px;
    line-height: 33px;
    margin: 0 0 20px 0;
  }

  .c1 p {
    font-size: 20px;
    line-height: 23px;
    margin: 0 0 20px 0;
  }
}

@media screen and (min-width:1200px) {
  .c1 h2 {
    font-weight: 500;
    font-size: 40px;
    line-height: 47px;
  }

  .c1 p {
    font-weight: 500;
    font-size: 28px;
    line-height: 33px;
  }
}

<div>
    <li
      class="c0"
    >
      <img />
      <div
        class="c1"
      >
        <h2 />
        <p>
          Price: 
          $
        </p>
        <div
          class="c2"
        >
          <p>
            Amount: 
          </p>
        </div>
        <div
          class="c3"
          data-testid="delete-btn"
        >
          <svg
            class="trash"
          >
            <use
              xlink:href="sprite.svg#trash"
            />
          </svg>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
